lib/rex/proto/http/client.rb

```diff
@@ -234,10 +237,11 @@ def _send_recv(req, t = -1, persist = false)
     elsif req.respond_to?(:opts) && req.opts['krb_transform_request'] && self.krb_encryptor
       req = req.opts['krb_transform_request'].call(self.krb_encryptor, req)
     end
-
+    subscriber.on_request(req)
     send_request(req, t)
 
     res = read_response(t, :original_request => req)
+    subscriber.on_response(res)
```

Comment:
We should move this after the decryption happens:

```ruby
    if req.respond_to?(:opts) && req.opts['ntlm_transform_response'] && self.ntlm_client
      req.opts['ntlm_transform_response'].call(self.ntlm_client, res)
    elsif req.respond_to?(:opts) && req.opts['krb_transform_response'] && self.krb_encryptor
      req = req.opts['krb_transform_response'].call(self.krb_encryptor, res)
    end
    subscriber.on_response(res)
```

For the same reason as above; if logging happens immediately, it's encrypted, if it happens afterwards it's decrypted